wlcreate/Start-Asian-Love
# Refactor CSS for News and Support pages and continuing issue 13

## Description

Related Issue: #28 

This PR continues to refactor the CSS of the first version of Start Asian Love, specifically the News and Support pages. Officially the Home, Header, Footer, News, and Support pages/components are completed. The Wins and WinsId pages are not finished, the main change is the updated CSS grid to be responsive instead of specifying 3 columns for the `Wins Resources` component. Therefore, both pages need to be updated with global variables (fonts, colors, etc.), tweaked for responsiveness, and updated classnames/html.

Furthermore, something I noticed this time around is that when using Next.js' Image component, in addition to specifying the width and height in line, if you are getting the image from a 3rd party resource you will need to add the domain(s) via `next.config.js`. Since the news api is grabbing from many different domains I decided not to change the img to using the Image component, however we can still implement this in the future.

Also I officially added Sandy to the Readme and the Footer! 

*update as of Thurs, Jun 3:
I abstracted the allyshipList and CategoryList components into a reusableButton component since the two were essentially doing the same with the same data. Some conditional logic needed to be implement to know what the buttons are being generated for since the data is the same for both with the difference being whether to grab `allyship` vs `category`. Therefore at the moment as I pass down as props the data and associated onClick event to change the selection, I am also passing down which `group` it belongs to.

Additionally, I added some more text content to the Support page to introduce the user to the page more and changed the `img` for the Resource card to use Next.js'. The transition to `Image` continues on what I mentioned in my first draft of this PR; I added the `next.config.js` with the domain for `res.cloudinary.com` since that is where a lot of our images are hosted.

*update as of Fri, Jun 4:
I noticed when checking our [PageSpeed Insights](https://developers.google.com/speed/pagespeed/insights/) that our home page did not have good results mainly because of the `react-player` package. After some consideration and realization that the package isn't doing anything I can't do without it, I decided to remove it and replicate the same behavior with vanilla HTML/CSS. While there is still room for improvement (before our first deployed version of SAL's home page score on mobile was 20 compared to 22 now and the desktop remains the same of 73), I do see the measured difference in the details.

Other future notes:
* be intentional with what each font is for (currently have around 5)
* know the breakpoints for responsive --> what size are we designing in mind first?
* create a general image for news articles when there is no image available

Resources:


* [CSS grid for justify items](https://www.digitalocean.com/community/tutorials/css-align-justify)
* [Complete Guide to CSS Grid](https://css-tricks.com/snippets/css/complete-guide-grid/)
* [Responsive Grid](https://css-tricks.com/look-ma-no-media-queries-responsive-layouts-using-css-grid/)
* [Cloudinary with NextJS](https://spacejelly.dev/posts/how-to-use-cloudinary-images-in-next-js-with-blurred-placeholders/)
* [`next.config.js` Information](https://nextjs.org/docs/api-reference/next.config.js/environment-variables)
* [Responsive Youtube Video Embeds](https://flaviocopes.com/responsive-youtube-videos/)
* [uninstalling npm packages](https://flaviocopes.com/npm-uninstall-packages/)

## Acceptance Criteria


- Reference suggestions from [PageSpeed Insights](https://developers.google.com/speed/pagespeed/insights/)
- Image Optimization (using Nextjs built in [Image component](https://nextjs.org/docs/basic-features/image-optimization))
- Include specific images to render based on the media query (re-size images for what we need)
- Defer offscreen images

## Self-check

Please check all that apply:



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|    | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|  ✓  | :scroll: Docs              |

## Updates

### Before


<img width="1430" alt="Screen Shot 2021-06-03 at 8 20 27 PM" src="https://user-images.githubusercontent.com/62153993/120727679-26899500-c4a9-11eb-9391-d2f004b90b4b.png">

<img width="1430" alt="Screen Shot 2021-06-03 at 8 21 04 PM" src="https://user-images.githubusercontent.com/62153993/120727710-3bfebf00-c4a9-11eb-8ab5-234ff40f35b5.png">

<img width="1430" alt="Screen Shot 2021-06-03 at 8 21 29 PM" src="https://user-images.githubusercontent.com/62153993/120727726-4a4cdb00-c4a9-11eb-9be2-21b1779661c6.png">

<img width="1430" alt="Screen Shot 2021-06-03 at 8 21 48 PM" src="https://user-images.githubusercontent.com/62153993/120727744-55077000-c4a9-11eb-8185-325431f535c7.png">

### After


<img width="1430" alt="Screen Shot 2021-06-03 at 8 22 09 PM" src="https://user-images.githubusercontent.com/62153993/120727767-62245f00-c4a9-11eb-92f7-17aa963493ec.png">

<img width="1430" alt="Screen Shot 2021-06-03 at 8 22 21 PM" src="https://user-images.githubusercontent.com/62153993/120727790-694b6d00-c4a9-11eb-81b3-4ca45beebd19.png">

<img width="1430" alt="Screen Shot 2021-06-03 at 8 22 34 PM" src="https://user-images.githubusercontent.com/62153993/120727798-710b1180-c4a9-11eb-87c2-3594f96f6507.png">

<img width="1430" alt="Screen Shot 2021-06-03 at 8 22 52 PM" src="https://user-images.githubusercontent.com/62153993/120727805-7b2d1000-c4a9-11eb-92ac-10e8b78a21e7.png">

## Testing Steps / QA Criteria


* switch to this branch or view the [preview](https://start-asian-love-ayhrff3w7-waverleynyc-gmailcom.vercel.app/)
* confirm if there's anything strange in the responsive behavior (phone, tablet, laptop/desktop) for the different pages
